flax/experimental/nnx/nnx/graph_utils.py

```diff
@@ -746,23 +756,32 @@ def _graph_update_dynamic(
 
     # case 2: subgraph is being updated
     if is_node(current_value):
-      if isinstance(value, Variable):
+      if is_state_leaf(value):
         raise ValueError(
           f'Expected a subgraph for {key!r}, but got a Variable: {value!r}'
         )
+      assert not isinstance(value, Variable)
```

Comment:
Is this `assert` only needed to please the type checker?

Reply:
indeed